aws/resource_aws_codebuild_project.go

```diff
@@ -177,6 +177,10 @@ func resourceAwsCodeBuildProject() *schema.Resource {
 							Optional: true,
 							Default:  false,
 						},
+						"certificate": {
```

Comment:
Since the CodeBuild API seems to validate against `.pem` and `.zip` endings, we can implement plan-time validation for that via:

```go
ValidateFunc: validation.StringMatch(regexp.MustCompile(`\.(pem|zip)$`), "must end in .pem or .zip"),
```

Also, since this attribute lives inside a `Type: schema.TypeSet` attribute, there is a `Set` function which is used to detect changes. `resourceAwsCodeBuildProjectEnvironmentHash` in this case.

We can add the following there:

```go
if v, ok := m["certificate"]; ok && v.(string) != "" {
	buf.WriteString(fmt.Sprintf("%s-", v.(string)))
}
```

At some point we'll likely be converting many of these `TypeSet` attributes, due to their unnecessary complexity.